vowpalwabbit/config/options.cc

```diff
@@ -77,7 +78,10 @@ bool options_i::add_parse_and_check_necessary(const option_group_definition& gro
     m_options[option->m_name] = option;
   }
 
-  return group.check_necessary_enabled(*this) && group.check_one_of();
+  auto necessary_enabled = group.check_necessary_enabled(*this);
+  if (necessary_enabled) { group.check_one_of(); }
```

Comment:
Are there not groups without any `necessary` options that we still want to check one_of, Eg most of the groups in parse_args

Reply:
Yeah they are checked in the other `add_and_parse` function. In the case of necessary we only check the `one_of` if the group was enabled. This is due to `cb_type` having different sets of allowed values.